### PR TITLE
1779 extinguisher cabinet

### LINKED ISSUE #1779: Extinguisher lockers
_Originally posted by @SoulSloth in https://github.com/space-wizards/space-station-14/issues/1575#issuecomment-675566151_

### PR DESCRIPTION
**Pull request by StrawberryMoses**
Fix #1779 

Adds fire extinguisher lockers, but at the request of A they were not mapped in
extinguisher

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
So 

Wouldn't you want something like this to be global instead of making it specifically  for the fire extinguisher? There's quite a lot of "single item containers" in SS13. 

Display cases, fireaxe wall storages... uhhh there's some other stuff i'm sure.

**Comment by RemieRichards** [comment]
@SweptWasTaken I think you mean *generalised*, not global.
But yes, this could potentially be made into a simple 'SingleItemStorage' component.

Not sure if it's the best idea though. Prevents us from having more unique visualisations though (such as nice unshaded lights on things like display cases)

**Comment by StrawberryMoses** [comment]
Yeah, I'm not sure having a generalized component will save us any work tbh... even in just this case, these cabinets are supposed to  accept mini fire extinguishers as well

File: Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            if (_opened)
+            {
+                if (ItemContainer.ContainedEntity == null)
+                {
+                    _opened = false;
+                    ClickLatchSound();
+                }
+                else if (eventArgs.User.TryGetComponent(out HandsComponent hands))
+                {
+                    _notifyManager.PopupMessage(Owner.Transform.GridPosition, eventArgs.User,
```
Comment (DrSmugleaf) [inline Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs:73]: Use Owner.PopupMessage here instead.

File: Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            {
+                appearance.SetData(ExtinguisherCabinetVisuals.IsOpen, _opened);
+                appearance.SetData(ExtinguisherCabinetVisuals.ContainsExtinguisher, ItemContainer.ContainedEntity != null);
+            }
+        }
+
+        private void ClickLatchSound()
+        {
+            EntitySystem.Get<AudioSystem>() // Don't have original click, this sounds close
+                .PlayFromEntity("/Audio/Machines/machine_switch.ogg", Owner, AudioHelpers.WithVariation(0.15f));
```
Comment (DrSmugleaf) [inline Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs:112]: This should be made into a property in ExposeData

Comment (StrawberryMoses) [reply Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs:112]: the .ogg file name? Expose data as in variable viewer thingy?

Comment (DrSmugleaf) [reply Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs:112]: https://github.com/space-wizards/space-station-14/blob/master/Content.Server/GameObjects/Components/Strap/StrapComponent.cs#L136

File: Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs (hunk 1 above)
Comment (StrawberryMoses) [reply Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs]: done

File: Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs (hunk 2 above)
Comment (StrawberryMoses) [reply Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs]: done

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**